primitives/manta/src/xcm.rs

```diff
@@ -365,8 +365,15 @@ impl<
 
         let (asset_id, amount, who) = Self::match_asset_and_location(asset, location)?;
 
-        MultiAdapterFungibleLedger::mint(asset_id, &who, amount)
-            .map_err(|_| XcmError::FailedToTransactAsset("Failed Mint"))?;
+        // If it's non-native asset we want to check with increase in total supply.
+        // Otherwise it will just use false, as it is assumed the native asset supply cannot be changed.
+        MultiAdapterFungibleLedger::can_deposit(asset_id, &who, amount, true).map_err(|_| {
```

Comment:
Just feel a little confused about line `370` and `374`. 
If line 370 receives non-native asset id, total supply will be increased, right?
But line 374 seems will increase total supply as well for non native asset.

Perhaps I don't understand related code here.

Reply:
Line 370 will check by increase total supply only for the non-native asset. Internally you can see that the native asset has `false` hard-coded and doesn't use the argument.

Line 374 will again only increase the total supply for non-native assets. For the native asset it will use `deposit_create` internally which doesn't increase the total supply.